### ISSUE TITLE
Log files piling up the home folder.

### ISSUE THREAD

**Issue by kelepiradam** [igvteam/igv#885]
Hi 

I noticed that recently IGV has started to keep logs in the home folder and if not tracked they seem to pileup quickly. Is it possible to add a switch to disable logging if it is not absolutely necessary?

Regards.

**Comment by jrobinso** [comment]
There should be a log4j switch to control this, actually it shouldn't be piling up but perhaps something has changed.    @davideby can "java_arguments" be used to disable log4j logging, or restrict it to a single file of max size?   

@kelepiradam Please report the version with any issues,  there are probably 50 extant versions at least still in use.

**Comment by kelepiradam** [comment]
I am on the latest version but this was an issue since the last 2 3 versions.

**Comment by davideby** [comment]
Hi @kelepiradam,

I'm looking into this today.  Could you confirm a few things for me?
- Which platform are you using (e.g. Windows 10, macOS Catalina, a particular Linux distribution)?
- Did you install IGV directly from our website or is it coming from somewhere else? Are you using the bundle with Java included or do you use a system-installed Java?
- Do you happen to do batch runs with IGV or leave it running for long periods of time?
- Do you have any other sense of what patterns of IGV use might trigger the log activity?

To answer your question @jrobinso, yes, it looks like we could provide an configuration override that the user could specify through the java_arguments file.  I need to check out exactly how that works.

Obviously it would also be good to cut down on the overall logging as well.

**Comment by kelepiradam** [comment]
Hi @davideby 

Sure
I am using Fedora 33.
I often check the main download page from the main IGV website and download the latest binary. I tend not to use the bundled java and use the platform native openjdk. I don't perform batch runs but I tend to leave IGV open for long periods of time but the log files show up and update whenever I navigate different loci. Logs are often filled with navigation clues. 

```
INFO [2021-01-06T23:02:47,605] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,605] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,606] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,687] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,687] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,689] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,688] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,690] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,690] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,691] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,691] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
INFO [2021-01-06T23:02:47,691] [AlignmentDataManager.java:339]  Already loading: chr17:7666401-7689550
```

One interesting thing is the beginning of the log file seems to be originating from the previous session. For example this log file was not there last time I checked and when I loaded a cram file and navigated to TP53 locus file popped up. Normally you would expect to have the beginning of the file to contain startup messages however the beginning of the log contains INFO related to another loci which I have not visited during this session. I am wondering where these logs are kept before being pushed to the home folder log file.

**Comment by davideby** [comment]
Thanks @kelepiradam, that's helpful.  Another question, then: do these files seem to be limited by size?

@jrobinso, I wonder if maybe some of these log message might be more appropriate at DEBUG level?

**Comment by kelepiradam** [comment]
They are indeed. They don't seem to exceed 1KB size and they all have the same size.

**Comment by jrobinso** [comment]
That doesn't look like a released build, if it was it was a mistake to have
that logging statement there.   When you say "latest binary" are you
referring to a released package or a development snapshot?

**Comment by davideby** [comment]
Hmm, very strange.  1 KB is awfully small.  @kelepiradam, did you mean 1 *MB*?  In my tests with the latest build on Linux Mint, the log rolls over at 1 MB.  But it just keeps using `igv.log` and wipes out the previous contents so it never grows beyond 1 MB and doesn't create any "backup" logs.

Can I ask what system-level Java you are using?  Could you try `java -showversion` at the command line?  Also, how do you launch IGV?

@jrobinso, FWIW I'm seeing those same INFO messages in my tests on 2.8.13.

**Comment by kelepiradam** [comment]
Sorry 1MB each. 

This is the version that I am using. Downloaded from IGV webpage downloads section directly. 

![image](https://user-images.githubusercontent.com/37572619/103818452-9f5eff00-5079-11eb-8a48-c622f3aa0582.png)

Here is the java version.

```
➜  ~ java -version
openjdk version "11.0.9.1" 2020-11-04
OpenJDK Runtime Environment 18.9 (build 11.0.9.1+11)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.9.1+11, mixed mode, sharing)
```

**Comment by davideby** [comment]
That all looks quite normal @kelepiradam.  Your version of Java is slightly newer than the one embedded in our "With Java" bundles but that should not affect anything (in theory anyway).  Perhaps there's a system-level logging configuration somewhere that affects things?  I'll need to investigate some more and see what I can find.

**Comment by davideby** [comment]
Hi @kelepiradam,

I haven't figured out what's going on with your system as I can't reproduce this myself.  However, we have a new option that should let you turn off logging altogether.

The basic idea is to override the default logging configuration (internal to the igv.jar file) with a [new external configuration file](https://raw.githubusercontent.com/igvteam/igv/master/src/main/resources/log4j2_none.xml).  The details are at the end of our [updated readme](https://raw.githubusercontent.com/igvteam/igv/master/scripts/readme.txt) but I'll summarize here.

The latest versions of IGV create a `.igv` directory containing a `java_arguments` file in the user's home directory.  Download that new external configuration file to the `.igv` directory, and then add a line like the following to `java_arguments` with a text editor like emacs or vi or similar:
   `-Dlog4j.configurationFile=/home/kelepiradam/.igv/log4j2-none.xml`

That file path is just an example and should obviously reflect your home directory instead.  I'd recommend using an absolute path and also **not** using `~` or any environment variables here just to avoid any issues with expansion.

Let me know how it goes.  There seems to be something different about your system, so it's possible this still won't work.  Still, this *should* do the job.

- David

**Comment by kelepiradam** [comment]
Thanks @davideby  and @jrobinso 

The new config file for log4j solved the logging issue. 

Thanks again.

**Comment by davideby** [comment]
@kelepiradam By the way, we found and fixed some related logging issues that will be included in the next release (2.9 series).  That *should* go out in the next few days, if you're interested in having more reasonable logging in place.
